# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../mozilla/gcc-4.0.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -Naur mozilla_orig/security/nss/lib/pki1/oiddata.h mozilla/security/nss/lib/pki1/oiddata.h
--- mozilla_orig/security/nss/lib/pki1/oiddata.h	2002-01-04 06:22:07.000000000 +0100
+++ mozilla/security/nss/lib/pki1/oiddata.h	2005-09-10 13:12:13.984991440 +0200
@@ -43,8 +43,8 @@
 #include "nsspki1t.h"
 #endif /* NSSPKI1T_H */
 
-extern const NSSOID nss_builtin_oids[];
-extern const PRUint32 nss_builtin_oid_count;
+/*extern const NSSOID nss_builtin_oids[];
+  extern const PRUint32 nss_builtin_oid_count; */
 
 /*extern const nssAttributeTypeAliasTable nss_attribute_type_aliases[];*/
 /*extern const PRUint32 nss_attribute_type_alias_count;*/
diff -Naur mozilla_orig/security/nss/lib/pki1/oidgen.perl mozilla/security/nss/lib/pki1/oidgen.perl
--- mozilla_orig/security/nss/lib/pki1/oidgen.perl	2001-11-08 01:15:25.000000000 +0100
+++ mozilla/security/nss/lib/pki1/oidgen.perl	2005-09-10 13:13:26.613950160 +0200
@@ -233,9 +233,6 @@
 #include "nsspki1t.h"
 #endif /* NSSPKI1T_H */
 
-extern const NSSOID nss_builtin_oids[];
-extern const PRUint32 nss_builtin_oid_count;
-
 /*extern const nssAttributeTypeAliasTable nss_attribute_type_aliases[];*/
 /*extern const PRUint32 nss_attribute_type_alias_count;*/
 
diff -Naur mozilla_orig/security/nss/lib/pki1/pki1.h mozilla/security/nss/lib/pki1/pki1.h
--- mozilla_orig/security/nss/lib/pki1/pki1.h	2001-07-19 22:40:42.000000000 +0200
+++ mozilla/security/nss/lib/pki1/pki1.h	2005-09-10 13:15:07.648590560 +0200
@@ -56,6 +56,8 @@
 PR_BEGIN_EXTERN_C
 
 /* fgmr 19990505 moved these here from oiddata.h */
+extern const NSSOID nss_builtin_oids[];
+extern const PRUint32 nss_builtin_oid_count;
 extern const nssAttributeTypeAliasTable nss_attribute_type_aliases[];
 extern const PRUint32 nss_attribute_type_alias_count;
 

Imported from Fedora Core.

  - Rene Rebe <rene@exactcode.de>

Index: netwerk/protocol/http/src/nsHttpConnectionMgr.cpp
===================================================================
RCS file: /cvsroot/mozilla/netwerk/protocol/http/src/nsHttpConnectionMgr.cpp,v
retrieving revision 1.7.20.1
diff -u -r1.7.20.1 nsHttpConnectionMgr.cpp
--- ./netwerk/protocol/http/src/nsHttpConnectionMgr.cpp	1 Sep 2004 23:31:03 -0000	1.7.20.1
+++ ./netwerk/protocol/http/src/nsHttpConnectionMgr.cpp	4 Mar 2005 19:59:57 -0000
@@ -834,8 +834,8 @@
 void
 nsHttpConnectionMgr::OnMsgUpdateParam(nsresult status, void *param)
 {
-    PRUint16 name  = (PRUint32(param) & 0xFFFF0000) >> 16;
-    PRUint16 value =  PRUint32(param) & 0x0000FFFF;
+    PRUint16 name  = (NS_PTR_TO_INT32(param) & 0xFFFF0000) >> 16;
+    PRUint16 value =  NS_PTR_TO_INT32(param) & 0x0000FFFF;
 
     switch (name) {
     case MAX_CONNECTIONS:
Index: widget/src/gtk2/nsDragService.cpp
===================================================================
RCS file: /cvsroot/mozilla/widget/src/gtk2/nsDragService.cpp,v
retrieving revision 1.5
diff -u -r1.5 nsDragService.cpp
--- ./widget/src/gtk2/nsDragService.cpp	30 Oct 2003 01:48:41 -0000	1.5
+++ ./widget/src/gtk2/nsDragService.cpp	4 Mar 2005 20:00:01 -0000
@@ -838,7 +838,7 @@
             (GtkTargetEntry *)g_malloc(sizeof(GtkTargetEntry));
         listTarget->target = g_strdup(gMimeListType);
         listTarget->flags = 0;
-        listTarget->info = (guint)listAtom;
+        listTarget->info = (guint)(NS_PTR_TO_INT32(listAtom));
         PR_LOG(sDragLm, PR_LOG_DEBUG,
                ("automatically adding target %s with id %ld\n",
                listTarget->target, listAtom));
@@ -877,7 +877,7 @@
                              (GtkTargetEntry *)g_malloc(sizeof(GtkTargetEntry));
                             listTarget->target = g_strdup(gTextUriListType);
                             listTarget->flags = 0;
-                            listTarget->info = (guint)listAtom;
+                            listTarget->info = (guint)NS_PTR_TO_INT32(listAtom);
                             PR_LOG(sDragLm, PR_LOG_DEBUG,
                                    ("automatically adding target %s with \
                                    id %ld\n", listTarget->target, listAtom));
@@ -914,7 +914,7 @@
                           (GtkTargetEntry *)g_malloc(sizeof(GtkTargetEntry));
                         target->target = g_strdup(flavorStr);
                         target->flags = 0;
-                        target->info = (guint)atom;
+                        target->info = (guint)NS_PTR_TO_INT32(atom);
                         PR_LOG(sDragLm, PR_LOG_DEBUG,
                                ("adding target %s with id %ld\n",
                                target->target, atom));
@@ -931,7 +931,7 @@
                              (GtkTargetEntry *)g_malloc(sizeof(GtkTargetEntry));
                             plainTarget->target = g_strdup(kTextMime);
                             plainTarget->flags = 0;
-                            plainTarget->info = (guint)plainAtom;
+                            plainTarget->info = (guint)NS_PTR_TO_INT32(plainAtom);
                             PR_LOG(sDragLm, PR_LOG_DEBUG,
                                    ("automatically adding target %s with \
                                    id %ld\n", plainTarget->target, plainAtom));
@@ -948,7 +948,7 @@
                              (GtkTargetEntry *)g_malloc(sizeof(GtkTargetEntry));
                             urlTarget->target = g_strdup(gMozUrlType);
                             urlTarget->flags = 0;
-                            urlTarget->info = (guint)urlAtom;
+                            urlTarget->info = (guint)NS_PTR_TO_INT32(urlAtom);
                             PR_LOG(sDragLm, PR_LOG_DEBUG,
                                    ("automatically adding target %s with \
                                    id %ld\n", urlTarget->target, urlAtom));

And last, but not least, a fix for 64bit systems.

  - Rene Rebe <rene@exactcode.de>

--- mozilla/gfx/src/freetype/nsFreeType.cpp.vanilla	2005-09-11 13:44:37.000000000 +0200
+++ mozilla/gfx/src/freetype/nsFreeType.cpp	2005-09-11 13:38:49.000000000 +0200
@@ -96,7 +96,7 @@
 // Define the FreeType2 functions we resolve at run time.
 // see the comment near nsFreeType2::DoneFace() for more info
 //
-#define NS_FT2_OFFSET(f) (int)&((nsFreeType2*)0)->f
+#define NS_FT2_OFFSET(f) (size_t)&((nsFreeType2*)0)->f
 FtFuncList nsFreeType2::FtFuncs [] = {
   {"FT_Done_Face",            NS_FT2_OFFSET(nsFT_Done_Face),            PR_TRUE},
   {"FT_Done_FreeType",        NS_FT2_OFFSET(nsFT_Done_FreeType),        PR_TRUE},
